engine/consensus/message_hub/message_hub.go

```diff
@@ -0,0 +1,504 @@
+	// We use by default 5 go-routines here. This is fine, because outbound messages are temporally sparse
+	// under normal operations. Hence, the go-routines should mostly be asleep waiting for work.
+	for i := 0; i < defaultMessageHubRequestsWorkers; i++ {
+		componentBuilder.AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			ready()
+			hub.queuedMessagesProcessingLoop(ctx)
+		})
+	}
+	hub.ComponentManager = componentBuilder.Build()
+	return hub, nil
+}
+
+// queuedMessagesProcessingLoop orchestrates dispatching of previously queued messages
+func (h *MessageHub) queuedMessagesProcessingLoop(ctx irrecoverable.SignalerContext) {
+	notifier := h.queuedMessagesNotifier.Channel()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-notifier:
+			err := h.processQueuedMessages(ctx)
+			if err != nil {
+				ctx.Throw(fmt.Errorf("internal error processing queued messages: %w", err))
+				return
+			}
+		}
+	}
+}
+
+// processQueuedMessages is a function which dispatches previously queued messages on worker thread
+// This function is called whenever we have queued messages ready to be dispatched.
+// No errors are expected during normal operations.
+func (h *MessageHub) processQueuedMessages(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+		}
+
+		msg, ok := h.queuedProposals.Pop()
+		if ok {
+			block := msg.(*flow.Header)
+			err := h.processQueuedProposal(block)
+			if err != nil {
+				return fmt.Errorf("could not process queued block %v: %w", block.ID(), err)
+			}
+
+			continue
+		}
+
+		msg, ok = h.queuedVotes.Pop()
+		if ok {
+			packed := msg.(*packedVote)
+			err := h.processQueuedVote(packed)
+			if err != nil {
+				return fmt.Errorf("could not process queued vote: %w", err)
+			}
+
+			continue
+		}
+
+		msg, ok = h.queuedTimeouts.Pop()
+		if ok {
+			err := h.processQueuedTimeout(msg.(*model.TimeoutObject))
+			if err != nil {
+				return fmt.Errorf("coult not process queued timeout: %w", err)
+			}
+
+			continue
+		}
+
+		// when there is no more messages in the queue, back to the loop to wait
+		// for the next incoming message to arrive.
+		return nil
+	}
+}
+
+// processQueuedTimeout performs actual processing of model.TimeoutObject, as a result of successful invocation
+// broadcasts timeout object to consensus committee.
+// No errors are expected during normal operations.
+func (h *MessageHub) processQueuedTimeout(timeout *model.TimeoutObject) error {
+	log := timeout.LogContext(h.log).Logger()
+
+	log.Info().Msg("processing timeout broadcast request from hotstuff")
+
+	// Retrieve all consensus nodes (excluding myself).
+	// CAUTION: We must include also nodes with weight zero, because otherwise
+	//          TCs might not be constructed at epoch switchover.
+	recipients, err := h.state.Final().Identities(filter.And(
+		filter.HasRole(flow.RoleConsensus),
+		filter.Not(filter.HasNodeID(h.me.NodeID())),
```

Comment:
I feel we could polish the filtering logic here:
* there is no need to send _anything_ to an ejected node, which I think we would do at the moment (?)

```suggestion
		filter.Not(filter.Ejected),
		filter.HasRole(flow.RoleConsensus),
		filter.Not(filter.HasNodeID(h.me.NodeID())),
```